tasks/jira/src/main/java/com/walmartlabs/concord/plugins/jira/TaskParams.java

```diff
@@ -102,17 +88,27 @@ public String jiraUrl() {
 
     @Override
     public long connectTimeout() {
-        return variables.getLong(CLIENT_CONNECTTIMEOUT, JiraClientCfg.super.connectTimeout());
+        return variables.getLong(CLIENT_CONNECT_TIMEOUT_KEY, JiraClientCfg.super.connectTimeout());
     }
 
     @Override
     public long readTimeout() {
-        return variables.getLong(CLIENT_READTIMEOUT, JiraClientCfg.super.readTimeout());
+        return variables.getLong(CLIENT_READ_TIMEOUT_KEY, JiraClientCfg.super.readTimeout());
     }
 
     @Override
     public long writeTimeout() {
-        return variables.getLong(CLIENT_WRITETIMEOUT, JiraClientCfg.super.writeTimeout());
+        return variables.getLong(CLIENT_WRITE_TIMEOUT_KEY, JiraClientCfg.super.writeTimeout());
+    }
+
+    @Override
+    public String userAgent() {
+        return Optional.ofNullable(variables.getString(USER_AGENT_KEY))
+                .orElseGet(() -> "Concord-Jira-Plugin: " + txId());
+    }
+
+    public String txId() {
+      return variables.assertVariable(TX_ID_KEY, Object.class).toString();
```

Comment:
```
context.processInstanceId()
```
or is this not a process identifier?

Reply:
Not sure I understand the question/suggestion here. The overall task behavior is to use the process id (`txId` process variable in both runtimes). However, in runtime-v2 `Context`, that value is explicitly available through `context..processInstanceId()`.

So in the task params merging (or just before), the value has to be copied into one of the input variable `Map`s or `TaskParams` would ned to be refactored to have another field for the process identifier--which seems a little unnecessary with the current big-bucket(map)-of-vars pattern.

It _is_ decoupled enough from most of the runtime bits so that `TaskParams` and `JiraTaskCommon` could possibly used on their own.

Reply:
sry. my bad. missed this part:
```
policyDefaults.put("txId", context.processInstanceId());
```

It seemed that `txId` needed to be explicitly passed as an input parameter.

Reply:
Yeah it's in the growing pains of a super simple task requiring more system variables/services over time. We could further refactor `JiraTaskCommon` to accept another config object that covers that sort of stuff, but not coupled with the runtime (e.g. container/supplier(s) for equivalent of `workDir`, `txId`, etc) so they don't appear to be expected `in:` params.